.ci/packer_cache.sh

```diff
@@ -1,3 +1,2 @@
 #!/usr/bin/env bash
-JAVA_HOME=$HOME/.java/java10
-./mvnw compile
+JAVA_HOME=$HOME/.java/java10 ./mvnw compile
```

Comment:
Don't we export vars for this like `JAVA10_HOME` to avoid hardcodes like `$HOME/.java/java10`? or am I thinking of something else?

Reply:
So like `JAVA_HOME=$JAVA10_HOME` would be a thing that I think/hope works

Reply:
Doesn't look like it... 

```
jenkins@elasticsearch-ci-immutable-ubuntu-1604-1539956431619082304:~$ env |grep JAVA
JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
jenkins@elasticsearch-ci-immutable-ubuntu-1604-1539956431619082304:~$ ls -la .java
total 44
drwxr-xr-x 11 jenkins jenkins 4096 Oct 18 14:51 .
drwxr-xr-x 16 jenkins jenkins 4096 Oct 19 14:34 ..
lrwxrwxrwx  1 jenkins jenkins   41 Oct 18 14:48 java10 -> /var/lib/jenkins/.java/oracle-10+46-linux
lrwxrwxrwx  1 jenkins jenkins   41 Oct 18 14:45 java11 -> /var/lib/jenkins/.java/oracle-11+28-linux
lrwxrwxrwx  1 jenkins jenkins   40 Oct 18 14:50 java7 -> /var/lib/jenkins/.java/oracle-7u80-linux
lrwxrwxrwx  1 jenkins jenkins   41 Oct 18 14:43 java8 -> /var/lib/jenkins/.java/oracle-8u192-linux
lrwxrwxrwx  1 jenkins jenkins   46 Oct 18 14:51 java8fips -> /var/lib/jenkins/.java/oracle-8u192-linux_fips
lrwxrwxrwx  1 jenkins jenkins   44 Oct 18 14:42 java9 -> /var/lib/jenkins/.java/oracle-9.0.4+11-linux
lrwxrwxrwx  1 jenkins jenkins   43 Oct 18 14:40 openjdk10 -> /var/lib/jenkins/.java/openjdk-10.0.2-linux
drwxr-xr-x  9 jenkins jenkins 4096 Oct 18 14:40 openjdk-10.0.2-linux
lrwxrwxrwx  1 jenkins jenkins   39 Oct 18 14:49 openjdk11 -> /var/lib/jenkins/.java/openjdk-11-linux
drwxr-xr-x  8 jenkins jenkins 4096 Oct 18 14:49 openjdk-11-linux
lrwxrwxrwx  1 jenkins jenkins   42 Oct 18 14:44 openjdk9 -> /var/lib/jenkins/.java/openjdk-9.0.4-linux
drwxr-xr-x  9 jenkins jenkins 4096 Oct 18 14:44 openjdk-9.0.4-linux
drwxr-xr-x  8 jenkins jenkins 4096 Oct 18 14:48 oracle-10+46-linux
drwxr-xr-x  8 jenkins jenkins 4096 Oct 18 14:45 oracle-11+28-linux
drwxr-xr-x  8 jenkins jenkins 4096 Oct 18 14:50 oracle-7u80-linux
drwxr-xr-x  7 jenkins jenkins 4096 Oct 18 14:43 oracle-8u192-linux
drwxr-xr-x  7 jenkins jenkins 4096 Oct 18 14:51 oracle-8u192-linux_fips
drwxr-xr-x  8 jenkins jenkins 4096 Oct 18 14:42 oracle-9.0.4+11-linux
```

Reply:
😞 that seems like a thing we should do then. Obviously I rescind any and all objection to this PR :)